src/renderer/services/file-management-system/index.ts

```diff
@@ -427,23 +429,30 @@ export default class FileManagementSystem {
           throw new Error("FileId was not published on COMPLETE upload: " + fssStatus.uploadId)
         }
         await this.complete(upload, fileId)
-      } else if (fssStatus.status === UploadStatus.WORKING) {
+      } else if (fssStatus.status === UploadStatus.RETRY || fssStatus.status === UploadStatus.WORKING) {
         // Update status to reflect the resume going smoothly
         await this.jss.updateJob(upload.jobId, {
           status: JSSJobStatus.RETRYING,
         });
-          // If FSS is still available to continue receiving chunks of this upload
-          // simply continue sending the chunks
-          let lastChunkNumber = fssStatus.chunkStatuses.findIndex(
-            (status) => status !== ChunkStatus.COMPLETE
-          );
-          if (lastChunkNumber === -1) {
-            lastChunkNumber = fssStatus.chunkStatuses.length;
-          }
-          if(!upload.serviceFields.md5CalculationInformation?.[`${lastChunkNumber}`]){
-            throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
-          }
 
+        // If FSS is still available to continue receiving chunks of this upload
+        // simply continue sending the chunks
+        let lastChunkNumber = fssStatus.chunkStatuses.findIndex(
+          (status) => status !== ChunkStatus.COMPLETE
+        );
+        if (lastChunkNumber === -1) {
+          lastChunkNumber = fssStatus.chunkStatuses.length;
+        }
+
+        const partiallyCalculatedMd5 = upload.serviceFields.md5CalculationInformation?.[`${lastChunkNumber}`];
+        if (!partiallyCalculatedMd5){
+          throw new Error('No partial MD5 for chunk ' + lastChunkNumber);
+        }
+
+        if (fssStatus.status === UploadStatus.RETRY ) {
```

Comment:
I moved `RETRY` & `WORKING` into a semi-combined though separated if block to de-duplicate the chunk status checking which is necessary for the deserialize (line 453). Lmk if that doesn't explain why I did it this way.

Reply:
Hm, im still confused.  Isn't the 452 block all that needs to happen on `RETRY`?  If so, it seems it could be a separate condition, and you could revert to the simpler condition on 432.

Anyway, i just wanted to know if you had overlooked an refactor opportunity.  Sounds like you have reasons for keeping it as is, which is fine w/me.

Reply:
We still need the `partiallyCalculatedMd5` variable that is defined in the outer if block